src/service/signup.service.ts

```diff
@@ -7,21 +7,44 @@ interface SignupData {
   uid: string;
 }
 
+interface CustomError extends Error {
+  code?: string;
+  target?: string;
```

Comment:
Prisma errors sometimes also contain the `errorCode` property (which is different from `code` for some confusing reason). It's probably good to include this too.